### PR TITLE
Fix MySQLUniqueViolation check to look for error dupe code more broadly

### PR DESCRIPTION
**Pull request by joshleckbee**
# Change
- Fixes MySQLUniqueViolation string matching to be less strict.

# Why
As of go-sql-driver mysql v1.7.0 the MySQL error code comes back as `Error {code} ({state}): {error message}`. Due to strict string matching the `MySQLUniqueViolation` func would fail to match since it looks for `Error {code}: {error message}`.

### CODE REVIEW

File: database/mysql.go [hunk 1]
```diff
@@ -199,7 +199,7 @@ func mysqlConnection(logger log.Logger, mysqlConfig *MySQLConfig, databaseName s
 // MySQLUniqueViolation returns true when the provided error matches the MySQL code
 // for duplicate entries (violating a unique table constraint).
 func MySQLUniqueViolation(err error) bool {
-	match := strings.Contains(err.Error(), fmt.Sprintf("Error %d: Duplicate entry", mySQLErrDuplicateKey))
+	match := strings.Contains(err.Error(), fmt.Sprintf("Error %d", mySQLErrDuplicateKey))
```
Comment (mhargrove) [inline database/mysql.go:202]: can we do the same for `mysqlErrDataTooLong` so it's less fragile too?

Comment (kolaworld) [inline database/mysql.go:202]: Worth doing a similar change for method `MySQLDataTooLong`?

File: database/mysql.go [hunk 2]
```diff
@@ -199,7 +199,7 @@ func mysqlConnection(logger log.Logger, mysqlConfig *MySQLConfig, databaseName s
...
 	if e, ok := err.(*gomysql.MySQLError); ok {
 		return match || e.Number == mySQLErrDuplicateKey
```
Comment (mhargrove) [inline database/mysql.go:204]: looks like [this](https://github.com/moov-io/base/blob/59aa06a665cd185868b743661723ef03d3857191/database/mysql.go#L47) is a dead link now, I think the new one would be https://dev.mysql.com/doc/mysql-errors/8.0/en/server-error-reference.html#error_er_dup_entry if we want to go ahead and update it while we're in here

File: database/mysql_test.go [hunk 3]
```diff
@@ -56,6 +56,13 @@ func TestMySQLUniqueViolation(t *testing.T) {
 	}
 }
 
+func TestMySQLUniqueViolation_WithStateValue(t *testing.T) {
+	err := errors.New(`problem upserting depository="282f6ffcd9ba5b029afbf2b739ee826e22d9df3b", userId="f25f48968da47ef1adb5b6531a1c2197295678ce": Error 1062 (23000): Duplicate entry '282f6ffcd9ba5b029afbf2b739ee826e22d9df3b' for key 'PRIMARY'`)
+	if !UniqueViolation(err) {
+		t.Error("should have matched unique violation")
+	}
+}
```
Comment (kolaworld) [inline database/mysql_test.go:64]: Following earlier comment, worth adding test method `TestMySQLDataTooLong_WithStateValue`?

File: database/mysql.go (hunk 1 above)
Comment (mhargrove) [reply database/mysql.go:202]: (@kolaworld beat me to it - resolving this as [dupe](https://github.com/moov-io/base/pull/265#discussion_r1038576034))

**Review by vxio — DISMISSED** [review]

File: database/mysql.go (hunk 2 above)
Comment (joshleckbee) [reply database/mysql.go:204]: Done

File: database/mysql.go (hunk 1 above)
Comment (joshleckbee) [reply database/mysql.go:202]: Done

File: database/mysql_test.go (hunk 3 above)
Comment (joshleckbee) [reply database/mysql_test.go:64]: Done

File: database/mysql.go (hunk 1 above)
Comment (joshleckbee) [reply database/mysql.go:202]: Done

**Review by kolaworld — APPROVED** [review]

**Review by mhargrove — APPROVED** [review]

**Review by InfernoJJ — APPROVED** [review]

**Pull request merged**